### PR TITLE
RFC: Add support for Checker Framework's nullable annotations

### PR DESCRIPTION
**Pull request by dpursehouse**

### CODE REVIEW

File: spotbugs/src/main/java/edu/umd/cs/findbugs/ba/NullnessAnnotation.java [hunk 1]
```diff
@@ -40,6 +40,8 @@ boolean match(@DottedClassName String className) {
                     || "com.google.common.base.Nullable".equals(className)
                     || "org.eclipse.jdt.annotation.Nullable".equals(className)
                     || "org.jetbrains.annotations.Nullable".equals(className)
+                    || "org.checkerframework.checker.nullness.qual.Nullable".equals(className)
```
Comment (iloveeclipse) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/ba/NullnessAnnotation.java]: I would remove all the `.Nullable".equals()` above and replace them all with `className.endsWith(".Nullable")`.

**Comment by dpursehouse** [comment]
@iloveeclipse thanks for the quick feedback.  I will retarget this to the 3.1 branch and address the comments.

I wasn't sure if tests need to be added for this?

**Comment by iloveeclipse** [comment]
> I wasn't sure if tests need to be added for this?

Good point, sorry I didn't mention this. Would be of course great if you could add a test.

**Comment by philipl** [comment]
You also need to update spotbugs/src/main/java/edu/umd/cs/findbugs/ba/jsr305/TypeQualifierResolver.java if you want the false positives in guava 23.6 to go away.

**Comment by dpursehouse** [comment]
Superseded by #539

**Pull request closed without merging**